src/main/java/org/folio/dcb/service/impl/BasicPickupLibraryServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.domain.dto.CirculationItemRequest;
+import org.folio.dcb.domain.dto.DcbTransaction;
+import org.folio.dcb.domain.dto.TransactionStatus;
+import org.folio.dcb.domain.dto.TransactionStatusResponse;
+import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.repository.TransactionRepository;
+import org.folio.dcb.service.CirculationItemService;
+import org.folio.dcb.service.CirculationService;
+import org.folio.dcb.service.LibraryService;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+import static org.folio.dcb.domain.dto.ItemStatus.NameEnum.AWAITING_PICKUP;
+import static org.folio.dcb.domain.dto.ItemStatus.NameEnum.CHECKED_OUT;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.CLOSED;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.CREATED;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.ITEM_CHECKED_IN;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.ITEM_CHECKED_OUT;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.OPEN;
+
+@Service("basicPickupLibraryService")
+@RequiredArgsConstructor
+@Log4j2
+public class BasicPickupLibraryServiceImpl implements LibraryService {
+
+  private final TransactionRepository transactionRepository;
+  private final CirculationService circulationService;
+  private final CirculationItemService circulationItemService;
+
+
+  @Override
+  public TransactionStatusResponse createCirculation(String dcbTransactionId, DcbTransaction dcbTransaction, String pickupServicePointId) {
+    return null;
+  }
+
+  @Override
+  public void updateStatusByTransactionEntity(TransactionEntity transactionEntity) {
+    log.debug("updateTransactionStatus:: Received checkIn event for itemId: {}", transactionEntity.getItemId());
+    CirculationItemRequest circulationItemRequest = circulationItemService.fetchItemById(transactionEntity.getItemId());
+    if (OPEN == transactionEntity.getStatus() && AWAITING_PICKUP == circulationItemRequest.getStatus().getName()) {
+      updateTransactionEntity(transactionEntity, TransactionStatus.StatusEnum.AWAITING_PICKUP);
+    } else if (TransactionStatus.StatusEnum.AWAITING_PICKUP == transactionEntity.getStatus() && CHECKED_OUT == circulationItemRequest.getStatus().getName()) {
+      updateTransactionEntity(transactionEntity, ITEM_CHECKED_OUT);
+    } else if(ITEM_CHECKED_OUT == transactionEntity.getStatus()){
+      log.info("updateStatusByTransactionEntity:: Updated item status from {} to {}",
```

Comment:
It would be nice to have these log info in every if/else if case as well. Can you please add them for every branch code of if/else if case of this method ?